perfmetrics/scripts/ls_metrics/listing_benchmark.py

```diff
@@ -0,0 +1,473 @@
+  num_contents = len(contents.decode('utf-8').split('\n')[:-1])
+  contents_url = contents.decode('utf-8').split('\n')[:-1]
+  return contents_url, num_contents
+
+
+def CompareDirectoryStructure(url, directory_structure) -> bool:
+  """Compares the directory structure present in the GCS bucket with the structure present in the JSON config file.
+
+  This function checks whether a same directory structure is present in the GCS bucket or not. If a same structure
+  is already present then we dont have to make the whole structure from scratch everytime. This saves a lot of time
+  as writing to the GCS bucket is a time expensive operation.
+
+  Args:
+    url: Path of the directory to compare in the GCS bucket.
+    directory_structure: Protobuf of the current directory.
+
+  Returns:
+    A boolean value which is true if the similar directory structure is already present in the GCS bucket.
+    Otherwise returns a false and we have to create the directory structure from scratch in the GCS bucket.
+  """
+  contents_url, num_contents = ListDirectory(url)
+
+  # gsutil in some cases return the contents_url list with the current directory in
+  # the first index. We dont want the current directory so we remove it
+  # manually.
+  if contents_url and contents_url[0] == url:
+    contents_url = contents_url[1:]
+
+  files = []
+  folders = []
+
+  for content in contents_url:
+    if content[-1] == '/':
+      folders.append(content)
+    else:
+      files.append(content)
+
+  if len(folders) != directory_structure.num_folders:
+    return False
+
+  if len(files) != directory_structure.num_files:
+    return False
+
+  result = True
+  for folder in directory_structure.folders:
+    new_url = url + folder.name + '/'
+    if new_url not in folders:
+      return False
+    result = result and CompareDirectoryStructure(new_url, folder)
+
+  return result
+
+
+def UnmountGCSBucket(gcs_bucket) -> None:
+  """Unmounts the GCS bucket.
+
+  Args:
+    gcs_bucket: Name of the directory to which GCS bucket is mounted to.
+
+  Raises:
+    An error if the GCS bucket cound not be umnounted.
+  """
+  log.info('Unmounting the GCS Bucket.\n')
+
+  exit_code = subprocess.call('umount -l {}'.format(gcs_bucket), shell=True)
+
+  if exit_code != 0:
+    log.error('Error encountered in umounting the bucket. Aborting!\n')
+    subprocess.call('bash', shell=True)
+
+  else:
+    subprocess.call('rm -rf {}'.format(gcs_bucket), shell=True)
+    log.info('Successfully unmounted the bucket and deleted %s directory.\n', gcs_bucket)
+
+
+def MountGCSBucket(bucket_name) -> str:
+  """Mounts the GCS bucket into the gcs_bucket directory.
+
+  Args:
+    bucket_name: Name of the bucket to be mounted.
+
+  Returns:
+    A string which contains the name of the directory to which the bucket is mounted.
+
+  Raises:
+    An error into the logs if cannot mount the bucket due to some error. Also shut offs the program.
```

Comment:
you need not mention about logging in the comments. just say aborts the program when mounting is unsuccessful.